test/metabase/query_processor_test.clj

```diff
@@ -418,33 +418,45 @@
     (is (= {:database 1, :type :query, :query {:source-query {:source-query {:native "wow"}}}}
            (nest-query {:database 1, :type :native, :native {:query "wow"}} 2)))))
 
-(defn do-with-bigquery-fks [d f]
-  (if-not (= driver/*driver* d)
-    (f)
-    (let [supports? driver/supports?]
-      (with-redefs [driver/supports? (fn [driver feature]
-                                       (if (= [driver feature] [d :foreign-keys])
-                                         true
-                                         (supports? driver feature)))]
-        (let [thunk (reduce
-                     (fn [thunk [source dest]]
-                       (fn []
-                         (tu/with-temp-vals-in-db Field (apply data/id source) {:fk_target_field_id (apply data/id dest)
-                                                                                :semantic_type      "type/FK"}
-                           (thunk))))
-                     f
-                     {[:checkins :user_id]   [:users :id]
-                      [:checkins :venue_id]  [:venues :id]
-                      [:venues :category_id] [:categories :id]})]
-          (thunk))))))
+(defn do-with-bigquery-fks [driver-or-drivers thunk]
+  {:pre [((some-fn keyword? coll?) driver-or-drivers)]}
+  (letfn [(add-fks? [driver]
+            (if (coll? driver-or-drivers)
+              (contains? (set driver-or-drivers) driver)
+              (= driver driver-or-drivers)))]
+    (if-not (add-fks? driver/*driver*)
+      (thunk)
+      (let [supports? driver/supports?]
+        (with-redefs [driver/supports? (fn [driver feature]
+                                         (if (and (add-fks? driver)
+                                                  (= feature :foreign-keys))
+                                           true
+                                           (supports? driver feature)))]
+          (let [thunk (reduce
+                       (fn [thunk [source dest]]
+                         (fn []
+                           (testing (format "With FK %s -> %s" source dest)
+                             (tu/with-temp-vals-in-db Field (apply data/id source) {:fk_target_field_id (apply data/id dest)
+                                                                                    :semantic_type      "type/FK"}
+                               (thunk)))))
+                       thunk
+                       (if (str/includes? (:name (data/db)) "sample")
+                         {[:orders :product_id]  [:products :id]
+                          [:orders :user_id]     [:people :id]
+                          [:reviews :product_id] [:products :id]}
+                         {[:checkins :user_id]   [:users :id]
+                          [:checkins :venue_id]  [:venues :id]
+                          [:venues :category_id] [:categories :id]}))]
+            (thunk)))))))
```

Comment:
I upgraded this so it works with the sample dataset and so we can use it on both `:bigquery` and `:bigquery-cloud-sdk` at the same time